src/udev/udev-builtin-blkid.c

```diff
@@ -236,21 +237,25 @@ static int builtin_blkid(struct udev_device *dev, int argc, char *argv[], bool t
         bool is_gpt = false;
 
         static const struct option options[] = {
-                { "offset", optional_argument, NULL, 'o' },
+                { "offset", required_argument, NULL, 'o' },
                 { "noraid", no_argument, NULL, 'R' },
                 {}
         };
 
         for (;;) {
                 int option;
 
-                option = getopt_long(argc, argv, "oR", options, NULL);
+                option = getopt_long(argc, argv, "o:R", options, NULL);
                 if (option == -1)
                         break;
 
                 switch (option) {
                 case 'o':
-                        offset = strtoull(optarg, NULL, 0);
+                        err = safe_atoi64(optarg, &offset);
+                        if (offset < 0)
+                                err = -ERANGE;
```

Comment:
hmm, we check `offset` now when it actually wasn't set by safe_atoi64() if that failed. Pleas rework to this:

```c
err = safe_atoi64(optarg, &offset);
if (err < 0)
        goto out;
if (offset < 0) {
        err = -ERANGE;
        goto out;
}
```

or some equivalent form.